shared/fs/filepreview/default-view.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import {globalStyles, globalColors, globalMargins} from '../../styles'
+import {Box, Button, Text} from '../../common-adapters'
+import PathItemInfo from '../common/path-item-info'
+import PathItemIcon from '../common/path-item-icon'
+import memoize from 'lodash/memoize'
+import {fileUIName, isMobile, isIOS} from '../../constants/platform'
+
+type DefaultViewProps = {
+  fileUIEnabled: boolean,
+  itemStyles: Types.ItemStyles,
+  pathItem: Types.PathItemMetadata,
+
+  onDownload: () => void,
+  onSave: () => void,
+  onShare: () => void,
+  onShowInFileUI: () => void,
+}
+
+const DefaultView = (props: DefaultViewProps) => (
+  <Box style={stylesContainer}>
+    <PathItemIcon spec={props.itemStyles.iconSpec} style={{}} />
+    <Text type="BodyBig" style={stylesFilename(props.itemStyles.textColor)}>
+      {props.pathItem.name}
+    </Text>
+    <Text type="BodySmall">{Constants.humanReadableFileSize(props.pathItem.size)}</Text>
+    {isMobile && (
+      <PathItemInfo
+        lastModifiedTimestamp={props.pathItem.lastModifiedTimestamp}
+        lastWriter={props.pathItem.lastWriter.username}
+        startWithLastModified={true}
+      />
+    )}
+    {isMobile && (
+      <Text type="BodySmall" style={stylesNoOpenMobile}>
+        This document can not be opened on mobile. You can still interact with it using the ••• menu.
+      </Text>
+    )}
+    {// Enable this button for desktop when we have in-app sharing.
+    isMobile && (
+      <Button
+        key="share"
+        type="Primary"
+        label="Share"
+        style={{marginTop: globalMargins.medium}}
+        onClick={props.onShare}
+      />
+    )}
+    {isIOS ? (
+      Constants.isImage(props.pathItem.name) && (
+        <Button
+          key="open"
+          type="Secondary"
+          label={'Save'}
+          style={{marginTop: globalMargins.small}}
+          onClick={props.onSave}
+        />
+      )
+    ) : props.fileUIEnabled ? (
+      <Button
+        key="open"
+        type="Secondary"
+        label={'Show in ' + fileUIName}
+        style={{marginTop: globalMargins.small}}
+        onClick={props.onShowInFileUI}
+      />
+    ) : (
+      <Button
```

Comment:
With the conditionals the way they are right now, this button will show up on Android. Is that desired?

Reply:
Cool, just making sure.